### PR TITLE
Reduce test time

### PR DESCRIPTION
**Pull request by okuta**

### CODE REVIEW

File: tests/chainer_tests/links_tests/connection_tests/test_n_step_lstm.py [hunk 1]
```diff
@@ -15,7 +15,6 @@ def sigmoid(x):
 
 
 @testing.parameterize(*testing.product({
-    'use_cudnn': ['always', 'auto', 'never'],
```
Comment (unnonouno) [inline tests/chainer_tests/links_tests/connection_tests/test_n_step_lstm.py]: Why did you remove it?

Comment (okuta) [inline tests/chainer_tests/links_tests/connection_tests/test_n_step_lstm.py]: This test case is redundant. 
`tests/chainer_tests/functionss_tests/connection_tests/test_n_step_lstm.py` is covered this case.

File: tests/chainer_tests/links_tests/connection_tests/test_n_step_rnn.py [hunk 2]
```diff
@@ -94,13 +93,12 @@ def check_forward(self, h_data, xs_data):
...
     def test_forward_cpu(self):
-        with chainer.using_config('use_cudnn', self.use_cudnn):
-            self.check_forward(self.h, self.xs)
+        self.check_forward(self.h, self.xs)
 
     @attr.gpu
     def test_forward_gpu(self):
         self.rnn.to_gpu()
-        with chainer.using_config('use_cudnn', self.use_cudnn):
+        with chainer.using_config('use_cudnn', 'auto'):
```
Comment (unnonouno) [inline tests/chainer_tests/links_tests/connection_tests/test_n_step_rnn.py]: Is 'always' better?

File: tests/chainer_tests/functions_tests/connection_tests/test_n_step_gru.py [hunk 3]
```diff
@@ -95,34 +92,27 @@ def check_forward(self, h_data, xs_data, ws_data, bs_data):
...
-    def test_forward_cpu_volatile(self):
-        with chainer.using_config('use_cudnn', self.use_cudnn), \
-                chainer.using_config('enable_backprop', False):
-            self.check_forward(self.hx, self.xs, self.ws, self.bs)
+        self.check_forward(self.hx, self.xs, self.ws, self.bs)
 
-    @attr.gpu
-    def test_forward_gpu(self):
-        with chainer.using_config('use_cudnn', self.use_cudnn), \
-                chainer.using_config('enable_backprop', True):
```
Comment (unnonouno) [inline tests/chainer_tests/functions_tests/connection_tests/test_n_step_gru.py]: I think this test is not required, but instead we need to test 'train' and 'test' mode.

Comment (okuta) [inline tests/chainer_tests/functions_tests/connection_tests/test_n_step_gru.py]: I fixed!

**Comment by unnonouno** [comment]
LGTM
